src/Microsoft.Framework.Runtime/ExportProviders/Resources/CreateCSharpManifestResourceName.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Globalization;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.Framework.Runtime
+{
+    internal static class CreateCSharpManifestResourceName
+    {
+        // Original source: https://raw.githubusercontent.com/Microsoft/msbuild/82177a50da735cc0443ac10fa490d69368403d71/src/XMakeTasks/CreateCSharpManifestResourceName.cs
+
+        public static string CreateManifestName(string fileName, string rootNamespace)
+        {
+            StringBuilder name = new StringBuilder();
+
+            // Differences from the msbuild task:
+            // - we do not include the name of the first class (if any) for binary resources or source code
+            // - culture info is ignored
+
+            if (rootNamespace != null && rootNamespace.Length > 0)
+            {
+                name.Append(rootNamespace).Append(".");
+            }
+
+            // Replace spaces in the directory name with underscores. Needed for compatibility with Everett.
+            // Note that spaces in the file name itself are preserved.
+            string path = MakeValidEverettIdentifier(Path.GetDirectoryName(fileName));
+
+            // This is different from the msbuild task: we always append extensions because otherwise,
+            // the emitted resource doesn't have an extension and it is not the same as in the classic
+            // C# assembly
+            name.Append(Path.Combine(path, Path.GetFileName(fileName)));
+            name.Replace(Path.DirectorySeparatorChar, '.');
+            name.Replace(Path.AltDirectorySeparatorChar, '.');
+
+            return name.ToString();
+        }
+
+        // The code below the same is same as here: https://raw.githubusercontent.com/Microsoft/msbuild/41b137cd8805079af7792995e044521d62fcb005/src/XMakeTasks/CreateManifestResourceName.cs
+
+        /// <summary>
+        /// This method is provided for compatibility with Everett which used to convert parts of resource names into
+        /// valid identifiers
+        /// </summary>
+        /// <param name="name"></param>
+        /// <returns></returns>
+        private static string MakeValidEverettIdentifier(string name)
```

Comment:
I know what "Everett" means here, but it might be good to either: a) put a comment above explaining what "Everett" means here or b) rename the methods to the actual VS version name.